### DISCUSSION TITLE
How to disable the auto-size feature?

### DISCUSSION THREAD

**Discussion by rsoika** [eclipse-glsp/glsp#862] [Q&A] [answered]
In the GLSP workflow example there a nice feature which allows editing the name of a node inline. Also after editing the name, the element resizes automatically to the with of the name.

![Peek 2023-01-13 15-49](https://user-images.githubusercontent.com/1339124/212348905-bef976c0-3d81-4a89-a3bc-1660f5cde7db.gif)

I did not found out from where this feature comes? I want to disable this behaviour, so that the size of a node did not change just because I edit the name. 

Can someone give me a hint from where this behaviour comes form?

**Comment by CamilleLetavernier** [comment, marked as answer]
Hi Ralph,

This should be related to the "resizeContainer" layout option on the label. This option defaults to true, so you may need to explicitly set it to false instead. However, note that labels don't support text ellipsis or line-wrap out of the box, so the text might leak outside of its parent with this option turned off.

HTH,
Camille

**Comment by rsoika** [comment]
Hi Camille,
ah OK, I played around with this option already on the parent container. But I was not aware that this option should be set to the label.

Finally I extended the edit label feature with the `ForeignObjectView` as explained [here](https://www.eclipse.org/glsp/documentation/rendering/). In this way the word wrap is now solved perfectly:

![Peek 2023-01-16 10-17](https://user-images.githubusercontent.com/1339124/212642285-5a0236ea-7a35-4815-8603-b841bb721a67.gif)

I also refined this feature for the labels outside of an element which are typical for BPMN. Here I use an auto-resize function to compute the size of the label on the server side. This also works now fine:
![Peek 2023-01-16 10-23](https://user-images.githubusercontent.com/1339124/212642944-37f14558-5dd3-4b34-bd7e-943b5ee165e3.gif)

GLSP is really a very powerful framework!
